### PR TITLE
feat: Added option to use GITUB_TOKEN

### PR DESCRIPTION
**Pull request by TomoakiMizuno**
image

ref:https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting

Due to rate limiting by GitHub REST API, 403 may be returned when retrieving releases.
As a countermeasure against rate limiting, requests can be sent using `GITHUB_TOKEN`.

### CODE REVIEW

File: src/setup.ts [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
+
+export async function getLatestVersion(token: string) {
   const url = "https://api.github.com/repos/loilo-inc/canarycage/releases";
-  type Release = {
-    tag_name: string;
-  };
-  const res = await fetch(url)
-  if (res.status === 200) {
-    const list: Release[] = await res.json()
+  if (token == "") {
```
Comment (keroxp) [inline src/setup.ts:12]: I think it's better to require tokens from next release.

**Review by keroxp — CHANGES_REQUESTED** [review]

File: src/setup.ts (hunk 1 above)
Comment (TomoakiMizuno) [reply src/setup.ts]: @keroxp Required github-token.

File: src/setup.ts [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+
+export async function getLatestVersion(token: string) {
   const url = "https://api.github.com/repos/loilo-inc/canarycage/releases";
-  type Release = {
-    tag_name: string;
-  };
-  const res = await fetch(url)
-  if (res.status === 200) {
-    const list: Release[] = await res.json()
+  const res = await getOctokit(token).request(`GET ${url}`);
```
Comment (keroxp) [inline src/setup.ts:11]: Please use the canonical way.
```suggestion
    const res = await getOctokit(token).rest.repos.listReleases({ owner: "loilo-inc", repo: "canarycage" })
```

**Review by keroxp — CHANGES_REQUESTED** [review]

File: src/setup.ts (hunk 2 above)
Comment (TomoakiMizuno) [reply src/setup.ts]: @keroxp [b1a82da](https://github.com/loilo-inc/actions-setup-cage/pull/20/commits/b1a82dae21ead7b45855c8b4091dd0deb1955ca5) fixed

**Review by keroxp — APPROVED** [review]

**Pull request merged**